From e6b8b498376797c20b514cce91e10f8fc3e2d3be Mon Sep 17 00:00:00 2001
From: Limeng <Meng.Li@windriver.com>
Date: Fri, 9 Aug 2019 06:58:46 +0000
Subject: [PATCH] driver: pci: pcie-rcar: fix build issue that is involed by
 upgrading kernel version

Recently, linux-yocto kernel veriosn upgrade from 4.18.40
to 4.18.41. When merge code into renesas-rcar branch with
commit 4c3e7553a960("PCI: rcar: Add the initialization of
PCIe link in resume_noirq()"), there is a conflict to
pcie driver, but not resolve properly.
Therefore, add this patch to fix it.

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/pci/controller/pcie-rcar.c | 18 ------------------
 1 file changed, 18 deletions(-)

diff --git a/drivers/pci/controller/pcie-rcar.c b/drivers/pci/controller/pcie-rcar.c
index c0b931eb532f..764a0fdee12e 100644
--- a/drivers/pci/controller/pcie-rcar.c
+++ b/drivers/pci/controller/pcie-rcar.c
@@ -1429,29 +1429,11 @@ static SIMPLE_DEV_PM_OPS(rcar_pcie_pm_ops,
 #define DEV_PM_OPS NULL
 #endif /* CONFIG_PM_SLEEP */
 
-static int rcar_pcie_resume_noirq(struct device *dev)
-{
-	struct rcar_pcie *pcie = dev_get_drvdata(dev);
-
-	if (rcar_pci_read_reg(pcie, PMSR) &&
-	    !(rcar_pci_read_reg(pcie, PCIETCTLR) & DL_DOWN))
-		return 0;
-
-	/* Re-establish the PCIe link */
-	rcar_pci_write_reg(pcie, CFINIT, PCIETCTLR);
-	return rcar_pcie_wait_for_dl(pcie);
-}
-
-static const struct dev_pm_ops rcar_pcie_pm_ops = {
-	.resume_noirq = rcar_pcie_resume_noirq,
-};
-
 static struct platform_driver rcar_pcie_driver = {
 	.driver = {
 		.name = "rcar-pcie",
 		.pm	= DEV_PM_OPS,
 		.of_match_table = rcar_pcie_of_match,
-		.pm = &rcar_pcie_pm_ops,
 		.suppress_bind_attrs = true,
 	},
 	.probe = rcar_pcie_probe,
-- 
2.17.1

